wurst/util/TerrainUtils.wurst

```diff
@@ -1,91 +1,149 @@
-	MoveLocation(tempLoc, x, y)
-	return GetLocationZ(tempLoc)
-	
-public function getTerrainZ(vec2 v) returns real
-	MoveLocation(tempLoc, v.x, v.y)
-	return GetLocationZ(tempLoc)
+public function vec2.getTerrainVariance() returns int
+	return GetTerrainVariance(this.x, this.y)
+
+/** Applies a single terrain deformation to closest tile. Is not immediate.
+	Reports shows that this may cause desync with Mac players. */
+public function vec2.addTerrainHeight(real val) returns terraindeformation
+	return TerrainDeformCrater(this.x, this.y, 63, -val, 1, true)
 
-public function vec2.getTerrainZ() returns real
-	MoveLocation(tempLoc, this.x, this.y)
-	return GetLocationZ(tempLoc)
+/** Fills 32x32 rect around vec2 with given pathing type */
+public function vec2.setTerrainPathable(pathingtype ttype, bool flag)
+	SetTerrainPathable(this.x, this.y, ttype, flag)
 	
-public function vec2.withTerrainZ() returns vec3
-	return vec3(this.x, this.y, getTerrainZ(this.x, this.y))
+public function vec2.isTerrainPathable(pathingtype ttype) returns bool
+	return IsTerrainPathable(this.x, this.y, ttype)
 	
-public function vec2.withTerrainZ(real zoffset) returns vec3
-	return vec3(this.x, this.y, getTerrainZ(this.x, this.y) + zoffset)
+/** Returns center of the Tile which given vec2 belongs to. */
+public function vec2.toTileCenter() returns vec2
+	return vec2(this.x.round(128), this.y.round(128))
 
-public function vec3.withTerrainZ() returns vec3
-	return vec3(this.x, this.y, getTerrainZ(this.x, this.y))
+/** Returns the (not normalised) terrain-normal at the given point */
+public function vec2.getTerrainNormal(real sampleRadius) returns vec3
+	return 2 * sampleRadius * vec3( (this.add(-sampleRadius, 0).getTerrainZ() - this.add(sampleRadius, 0).getTerrainZ()), (this.add(-sampleRadius, 0).getTerrainZ() - this.add(sampleRadius, 0).getTerrainZ()), 2*sampleRadius )
 	
-public function vec3.withTerrainZ(real zoffset) returns vec3
-	return vec3(this.x, this.y, getTerrainZ(this.x, this.y) + zoffset)
+public function vec2.isTerrainDeepWater() returns boolean
+	return not this.isTerrainPathable(PATHING_TYPE_FLOATABILITY) and this.isTerrainPathable(PATHING_TYPE_WALKABILITY)
+
+public function vec2.isTerrainShallowWater() returns boolean
+	return not this.isTerrainPathable(PATHING_TYPE_FLOATABILITY) and not this.isTerrainPathable(PATHING_TYPE_WALKABILITY) and this.isTerrainPathable(PATHING_TYPE_BUILDABILITY)
+
+public function vec2.isTerrainLand() returns boolean
+	return this.isTerrainPathable(PATHING_TYPE_FLOATABILITY)
+
+public function vec2.isTerrainPlatform() returns boolean
+	return not this.isTerrainPathable(PATHING_TYPE_FLOATABILITY) and not this.isTerrainPathable(PATHING_TYPE_WALKABILITY) and not this.isTerrainPathable(PATHING_TYPE_BUILDABILITY)
+
+constant	MAX_RANGE_SQ 	= 10.*10.
+constant 	DUMMY_ITEM_ID 	= 'wolg'
+let	   		dItem   		= CreateItem(DUMMY_ITEM_ID, 0, 0)..setVisible(false)
+let	   		find   			= Rect(0., 0., 128., 128.)
```

Comment:
these constants should adhere to naming convention. Change should be fine since they are not public.